File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cs
using System;

public class Solution {
    public int[] solution(int[] sequence, int k) {
        int[] answer = new int[2];
        
        int left = 0;
        int right = 0;
        
        int sum = 0;
        int size = sequence.Length;
        
        for(int i = right; i < sequence.Length; i++)
        {
            sum += sequence[i];
            while(right < sequence.Length && sum > k)
            {
                sum -= sequence[left];
                left++;
            }
            
            if(sum == k)
            {
                if(size > i - left)
                {
                    size = i - left;
                    answer[0] = left;
                    answer[1] = i;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/2/178870. 연속된 부분 수열의 합/연속된 부분 수열의 합.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> sequence, int k) {
    vector<int> answer;
    
    int left = 0;
    int sum = 0;
    int minVal = 1e9;
    
    for(int right = 0; right < sequence.size(); right++)
    {
        sum += sequence[right];
        
        while(sum > k)
        {
            sum -= sequence[left];
            left++;
        }
        
        if(sum == k)
        {
            int len = right - left;
            
            if(len < minVal)
            {
                minVal = len;
                answer = {left, right};
            }
        }
    }
    
    
    return answer;
}